### PR TITLE
Omis cancel order

### PR DESCRIPTION
**Pull request by marcofucci**
Add ability to cancel an order with `/v3/order/<order-pk>/cancel` and passing a cancellation reason.

### CODE REVIEW

File: datahub/omis/order/serializers.py [hunk 1]
```diff
@@ -206,6 +217,24 @@ def complete(self):
         return self.instance
 
 
+class CancelOrderSerializer(OrderSerializer):
+    """DRF serializer for cancelling an order."""
+
+    cancellation_reason = NestedRelatedField(CancellationReason, required=True)
```
Comment (reupen) [inline datahub/omis/order/serializers.py]: Isn't `required=True` the default?

File: datahub/omis/order/test/test_models.py [hunk 2]
```diff
@@ -551,6 +552,74 @@ def test_validation_error_if_not_all_actual_time_set(self):
...
+        Test that the order can be cancelled if it's in one of the allowed statuses.
+        """
+        reason = CancellationReason.objects.order_by('?').first()
+        order = OrderFactory(status=allowed_status)
+        adviser = AdviserFactory()
+
+        try:
+            with freeze_time('2018-07-12 13:00'):
+                order.cancel(by=adviser, reason=reason)
+        except Exception:
```
Comment (reupen) [inline datahub/omis/order/test/test_models.py]: Is this really needed? I mean the test will fail if it raises an exception and there are other assertions below.

File: datahub/omis/order/serializers.py (hunk 1 above)
Comment (marcofucci) [reply datahub/omis/order/serializers.py]: 👍

File: datahub/omis/order/test/test_models.py (hunk 2 above)
Comment (marcofucci) [reply datahub/omis/order/test/test_models.py]: hm maybe you are right

**Pull request merged**